hw/hw8/Hw8_2_sqlite3.py

```diff
@@ -0,0 +1,106 @@
+"""Homework 8.2."""
+import sqlite3
+from typing import Any, NoReturn, Tuple, Union
+
+
+class TableData:
+    """Wrap class TableData for database table.
+
+    When initialized with database name and table acts as collection object (implements Collection protocol)
+    example: instance = TableData(database_name='example.sqlite', table_name='presidents')
+    """
+
+    def __init__(self, database_name: str, table_name: str) -> None:
+        self.database_name = database_name
+        self.table_name = table_name
+        self.counter = 0
+        self.save = None
+
+    def __len__(self) -> int:
+        """Give current amount of rows in table in database.
+
+        example: len(instance)
+        Returns:
+            The return value. Amount of rows in table in database
+        """
+        with sqlite3.connect(self.database_name) as conn:
+            cursor = conn.cursor()
+        cursor.execute(f"SELECT count(*) from {self.table_name}")
+        return cursor.fetchone()[0]
+
+    def __contains__(self, item: str) -> bool:
+        """Check whether there is a database entry with the same name.
+
+        example: 'name' in instance
+        Args:
+            item: name
+
+        Returns:
+            The return value. True if available
+        """
+        with sqlite3.connect(self.database_name) as conn:
+            cursor = conn.cursor()
+        cursor.execute(
+            f"SELECT name from {self.table_name} where name=:name", {"name": item}
```

Comment:
It's better to use 'command' variable here and for others below to remove one level of nestation